spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,335 @@
+require 'spec_helper'
+
+describe ConferenceRegistrationsController, type: :controller do
+  let!(:first_user) { create(:user) }
+  let(:conference) { create(:conference) }
+  let(:conference_with_open_registration) { create(:conference) }
+  let!(:open_registration_period) { create(:registration_period, conference: conference_with_open_registration, start_date: Date.current - 6.days) }
+  let(:user) { create(:user) }
+
+  context 'user is not signed in' do
+    describe 'GET #new' do
+      context 'ichain is disabled' do
+        before do
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns conference, registration and user variable' do
+          expect(assigns(:conference)).to eq conference_with_open_registration
+          expect(assigns(:registration)).to be_instance_of Registration
+          expect(assigns(:user)).to be_instance_of User
+        end
+
+        it 'renders the new template' do
+          expect(response).to render_template('new')
+        end
+      end
+
+      context 'ichain is enabled' do
+        before do
+          CONFIG['authentication']['ichain']['enabled'] = true
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        after { CONFIG['authentication']['ichain']['enabled'] = false }
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+
+        it 'shows error in flash message' do
+          expect(flash[:alert]).to match 'You are not authorized to access this page. Maybe you need to sign in?'
+        end
+      end
+    end
+
+    describe 'POST #create' do
```

Comment:
@differentreality I have added it in context of `signed in user` [here](https://github.com/openSUSE/osem/pull/795/files#diff-f594354d23f55f0c31fc2d9fbd49250eR225). Do you want me to add the same thing here too?